cmake/FindARM.cmake

```diff
@@ -0,0 +1,67 @@
+# Check if the processor is an ARM and if Neon instruction are available on the machine where
+# the project is compiled.
+
+IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
+   EXEC_PROGRAM(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+
+   #neon instruction can be found on the majority part of modern ARM processor
+   STRING(REGEX REPLACE "^.*(neon).*$" "\\1" NEON_THERE ${CPUINFO})
+   STRING(COMPARE EQUAL "neon" "${NEON_THERE}" NEON_TRUE)
+   IF (NEON_TRUE)
+      set(NEON_FOUND true CACHE BOOL "NEON available on host")
```

Comment:
This and several other cache variable settings below seem strange to me--cache variables are usually for user-controlled input options, and it seems like many of the variables set here are not things users should see or specify themselves.

Rather, a user might control a cache variable that decides whether or not to *actually use* an option, while a regular CMake variable would store the information on whether or not that capability was found.